### DISCUSSION TITLE
Why does 'turbo-ignore' return an exit code, instead of a string?

### DISCUSSION THREAD

**Discussion by Tobjoern** [vercel/turborepo#5647] [Help] [answered]
From my current view, turbo-ignore returns the exit code '1' is a proejct changed, and '0' if it didn't.
Is there a specific reason behind this?

It leads to some very weird patterns, when integrating into a CI/CD system, epeciall with GH actions, since the immediately fail, whenvever a non-0 code is emitted.
But at the same time, I'm not very comfortable ignoring non-0 codes, since you never know, when something actually breaks.

![image](https://github.com/vercel/turbo/assets/31477665/40753255-9ecc-468c-982c-fa6335c2bf11)

**Comment by ntang86** [comment]
Hello, I have the same question ! 

I have a hard time to wrap my head around this... I mean, what's the logic behind that?
Is turbo-ignore suppose to be run in a CI/CD system? I guess yes, so why would this return an exit code 1 when I need to proceed to a deployment?

Regards

**Comment by ahriannn** [comment]
Want to bump this discussion, think it's kinda silly that there's not a flag to return a boolean or something instead of killing the entire process... @jaredpalmer think this could be looked into perhaps? I'd be happy to create a pull request with the change.

**Comment by mehulkar** [comment, marked as answer]
I think this is a good feature request, especially for `set -e`. Moved to issue https://github.com/vercel/turbo/issues/6348
